### PR TITLE
Improve test coverage for DPASAnalysis

### PR DESCRIPTION
**Pull request by etiotto**
This PR improves test coverage for DPASAnalysis by adding a null safety check for the minSGSizeAttr attribute and introducing comprehensive test files covering various DPAS scenarios including INT8 operations, edge cases, and alternative accumulation types.

### CODE REVIEW

File: test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xf32, #blocked>
+    %1 = ttg.convert_layout %result : tensor<128x128xf32, #blocked> -> tensor<128x128xf32, #blocked>
+    tt.store %arg2, %1 : tensor<128x128x!tt.ptr<f32>, #blocked>
+    tt.return
+  }
+}
+
+// -----
+
+// Test mismatched threads-per-warp and  min_sg_size (should NOT use DPAS)
```
[the remark this answers is not in the document]
Comment (etiotto) [reply test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/intel/intel-xpu-backend-for-triton/pull/6106#discussion_r2805590974)

File: test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  // CHECK-LABEL: tt.func public @kernel_fp16_to_fp16
+  tt.func public @kernel_fp16_to_fp16(
+    %arg0: tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>>,
+    %arg1: tensor<64x128xf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>>,
+    %arg2: tensor<128x128x!tt.ptr<f16>, #blocked>) {
+    %cst = arith.constant dense<0.000000e+00> : tensor<128x128xf16, #blocked>
+    // CHECK: tt.dot {{.*}} : tensor<128x64xf16{{.*}}> * tensor<64x128xf16{{.*}}> -> tensor<128x128xf16, #[[$DPAS]]>
+    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xf16, #blocked>
+    %1 = ttg.convert_layout %result : tensor<128x128xf16, #blocked> -> tensor<128x128xf16, #blocked>
+    tt.store %arg2, %1 : tensor<128x128x!tt.ptr<f16>, #blocked>
```
Comment (anmyachev) [inline test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir]: ?
```suggestion
    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xf16, #blocked>
    tt.store %arg2, %result : tensor<128x128x!tt.ptr<f16>, #blocked>
```

File: test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  // CHECK-LABEL: tt.func public @kernel_bf16_to_bf16
+  tt.func public @kernel_bf16_to_bf16(
+    %arg0: tensor<128x64xbf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>>,
+    %arg1: tensor<64x128xbf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>>,
+    %arg2: tensor<128x128x!tt.ptr<bf16>, #blocked>) {
+    %cst = arith.constant dense<0.000000e+00> : tensor<128x128xbf16, #blocked>
+    // CHECK: tt.dot {{.*}} : tensor<128x64xbf16{{.*}}> * tensor<64x128xbf16{{.*}}> -> tensor<128x128xbf16, #[[$DPAS2]]>
+    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xbf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xbf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xbf16, #blocked>
+    %1 = ttg.convert_layout %result : tensor<128x128xbf16, #blocked> -> tensor<128x128xbf16, #blocked>
+    tt.store %arg2, %1 : tensor<128x128x!tt.ptr<bf16>, #blocked>
```
Comment (anmyachev) [inline test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir]:
```suggestion
    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xbf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xbf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xbf16, #blocked>
    tt.store %arg2, %result : tensor<128x128x!tt.ptr<bf16>, #blocked>
```

File: test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  // CHECK-LABEL: tt.func public @kernel_fp16_bf16_mixed
+  tt.func public @kernel_fp16_bf16_mixed(
+    %arg0: tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>>,
+    %arg1: tensor<64x128xbf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>>,
+    %arg2: tensor<128x128x!tt.ptr<f32>, #blocked>) {
+    %cst = arith.constant dense<0.000000e+00> : tensor<128x128xf32, #blocked>
+    // Mixed FP16 and BF16 operands should not use DPAS
+    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xbf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xf32, #blocked>
+    %1 = ttg.convert_layout %result : tensor<128x128xf32, #blocked> -> tensor<128x128xf32, #blocked>
+    tt.store %arg2, %1 : tensor<128x128x!tt.ptr<f32>, #blocked>
```
Comment (anmyachev) [inline test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir]: ?
```suggestion
    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xbf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xf32, #blocked>
    tt.store %arg2, %result : tensor<128x128x!tt.ptr<f32>, #blocked>
```

File: test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  // CHECK-LABEL: tt.func public @kernel_no_dpas_support
+  tt.func public @kernel_no_dpas_support(
+    %arg0: tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>>,
+    %arg1: tensor<64x128xf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>>,
+    %arg2: tensor<128x128x!tt.ptr<f32>, #blocked>) {
+    %cst = arith.constant dense<0.000000e+00> : tensor<128x128xf32, #blocked>
+    // Without support attribute, should not use DPAS
+    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xf32, #blocked>
+    %1 = ttg.convert_layout %result : tensor<128x128xf32, #blocked> -> tensor<128x128xf32, #blocked>
+    tt.store %arg2, %1 : tensor<128x128x!tt.ptr<f32>, #blocked>
```
Comment (anmyachev) [inline test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir]:
```suggestion
    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xf32, #blocked>
    tt.store %arg2, %result : tensor<128x128x!tt.ptr<f32>, #blocked>
```

File: test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  // CHECK-LABEL: tt.func public @kernel_mismatched_warp_size
+  tt.func public @kernel_mismatched_warp_size(
+    %arg0: tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>>,
+    %arg1: tensor<64x128xf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>>,
+    %arg2: tensor<128x128x!tt.ptr<f32>, #blocked>) {
+    %cst = arith.constant dense<0.000000e+00> : tensor<128x128xf32, #blocked>
+    // threads-per-warp=32 doesn't match min_sg_size=16 without env var
+    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xf32, #blocked>
+    %1 = ttg.convert_layout %result : tensor<128x128xf32, #blocked> -> tensor<128x128xf32, #blocked>
+    tt.store %arg2, %1 : tensor<128x128x!tt.ptr<f32>, #blocked>
```
Comment (anmyachev) [inline test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir]:
```suggestion
    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xf16, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xf16, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xf32, #blocked>
    tt.store %arg2, %result : tensor<128x128x!tt.ptr<f32>, #blocked>
```

File: test/TritonIntelGPU/accelerate-matmul-int8.mlir [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  // CHECK-LABEL: tt.func public @kernel_int8_signed
+  tt.func public @kernel_int8_signed(
+    %arg0: tensor<128x64xi8, #ttg.dot_op<{opIdx = 0, parent = #blocked}>>,
+    %arg1: tensor<64x128xi8, #ttg.dot_op<{opIdx = 1, parent = #blocked}>>,
+    %arg2: tensor<128x128x!tt.ptr<i32>, #blocked>) {
+    %cst = arith.constant dense<0> : tensor<128x128xi32, #blocked>
+    // CHECK: tt.dot {{.*}} : tensor<128x64xi8{{.*}}> * tensor<64x128xi8{{.*}}> -> tensor<128x128xi32, #[[$DPAS]]>
+    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xi8, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xi8, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xi32, #blocked>
+    %1 = ttg.convert_layout %result : tensor<128x128xi32, #blocked> -> tensor<128x128xi32, #blocked>
+    tt.store %arg2, %1 : tensor<128x128x!tt.ptr<i32>, #blocked>
```
Comment (anmyachev) [inline test/TritonIntelGPU/accelerate-matmul-int8.mlir]:
```suggestion
    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xi8, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xi8, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xi32, #blocked>
    tt.store %arg2, %result : tensor<128x128x!tt.ptr<i32>, #blocked>
```

File: test/TritonIntelGPU/accelerate-matmul-int8.mlir [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  // CHECK-LABEL: tt.func public @kernel_int8_alt_config
+  tt.func public @kernel_int8_alt_config(
+    %arg0: tensor<128x64xi8, #ttg.dot_op<{opIdx = 0, parent = #blocked}>>,
+    %arg1: tensor<64x128xi8, #ttg.dot_op<{opIdx = 1, parent = #blocked}>>,
+    %arg2: tensor<128x128x!tt.ptr<i32>, #blocked>) {
+    %cst = arith.constant dense<0> : tensor<128x128xi32, #blocked>
+    // CHECK: tt.dot {{.*}} : tensor<128x64xi8{{.*}}> * tensor<64x128xi8{{.*}}> -> tensor<128x128xi32, #[[$DPAS2]]>
+    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xi8, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xi8, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xi32, #blocked>
+    %1 = ttg.convert_layout %result : tensor<128x128xi32, #blocked> -> tensor<128x128xi32, #blocked>
+    tt.store %arg2, %1 : tensor<128x128x!tt.ptr<i32>, #blocked>
```
Comment (anmyachev) [inline test/TritonIntelGPU/accelerate-matmul-int8.mlir]:
```suggestion
    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xi8, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xi8, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xi32, #blocked>
    tt.store %arg2, %result : tensor<128x128x!tt.ptr<i32>, #blocked>
```

File: test/TritonIntelGPU/accelerate-matmul-int8.mlir [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  // CHECK-LABEL: tt.func public @kernel_int8_wrong_output
+  tt.func public @kernel_int8_wrong_output(
+    %arg0: tensor<128x64xi8, #ttg.dot_op<{opIdx = 0, parent = #blocked}>>,
+    %arg1: tensor<64x128xi8, #ttg.dot_op<{opIdx = 1, parent = #blocked}>>,
+    %arg2: tensor<128x128x!tt.ptr<i16>, #blocked>) {
+    %cst = arith.constant dense<0> : tensor<128x128xi16, #blocked>
+    // INT8 -> INT16 not supported by DPAS (needs INT32 accumulator)
+    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xi8, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xi8, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xi16, #blocked>
+    %1 = ttg.convert_layout %result : tensor<128x128xi16, #blocked> -> tensor<128x128xi16, #blocked>
+    tt.store %arg2, %1 : tensor<128x128x!tt.ptr<i16>, #blocked>
```
Comment (anmyachev) [inline test/TritonIntelGPU/accelerate-matmul-int8.mlir]:
```suggestion
    %result = tt.dot %arg0, %arg1, %cst : tensor<128x64xi8, #ttg.dot_op<{opIdx = 0, parent = #blocked}>> * tensor<64x128xi8, #ttg.dot_op<{opIdx = 1, parent = #blocked}>> -> tensor<128x128xi16, #blocked>
    tt.store %arg2, %result : tensor<128x128x!tt.ptr<i16>, #blocked>
```

File: test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    %c = arith.constant dense<3.00e+00> : tensor<128x128xf32>
+    %d = tt.dot %a, %b, %c : tensor<128x64xf16> * tensor<64x128xbf16> -> tensor<128x128xf32>
+    tt.store %arg0, %d : tensor<128x128x!tt.ptr<f32>>
+    tt.return
+  }
+}
+
+// -----
+
+// RUN: triton-opt %s --split-input-file -triton-annotate-module='min-sg-size=8 support-2d-block-io=true support-dpas=true threads-per-warp=32' | FileCheck %s --check-prefix=CHECK-SG8
```
Comment (whitneywhtsang) [inline test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir]: didn't know you can have run line in the middle of a lit. How does it work? Does the run line in the beginning of the file still get run, does this run line run for this test case only or every test cases after?

File: test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+    %c = arith.constant dense<3.00e+00> : tensor<128x128xf32>
+    %d = tt.dot %a, %b, %c : tensor<128x64xf16> * tensor<64x128xf16> -> tensor<128x128xf32>
+    tt.store %arg0, %d : tensor<128x128x!tt.ptr<f32>>
+    tt.return
+  }
+}
+
+// -----
+
+// RUN: triton-opt %s --split-input-file -triton-annotate-module='min-sg-size=16 support-dpas=false threads-per-warp=16' | FileCheck %s --check-prefix=CHECK-NO-DPAS
```
Comment (whitneywhtsang) [inline test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir]: Should we add support-2d-block-io=true to make only support-dpas be the only changing factor?

Comment (whitneywhtsang) [reply test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir]: Should we not use threads-per-warp=16, as threads-per-warp would be 16 no matter support-dpas is true or false.

File: test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir [hunk 12]
```diff
@@ -0,0 +1,115 @@
+// RUN: triton-opt %s --split-input-file -triton-annotate-module='min-sg-size=16 support-2d-block-io=true support-dpas=true threads-per-warp=32' | FileCheck %s
+
+// Test that module annotations are applied correctly without dot operations
+// CHECK: module attributes {"ttg.threads-per-warp" = 32 : i32{{.*}}ttig.min_sg_size = 16
```
Comment (whitneywhtsang) [inline test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir]: why do we need to test `ttig.min_sg_size = 16`?

File: test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    tt.return
+  }
+}
+
+// -----
+
+// Test multiple dot operations in a function - all must be DPAS-compatible
```
Comment (whitneywhtsang) [inline test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir:41]: I don't see multiple dot operations in kernel_multiple_dots

File: test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir (hunk 2 above)
Comment (etiotto) [reply test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir]: done

File: test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir (hunk 3 above)
Comment (etiotto) [reply test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir]: done

File: test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir (hunk 4 above)
Comment (etiotto) [reply test/TritonIntelGPU/accelerate-matmul-alt-accum.mlir]: done

File: test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir (hunk 5 above)
Comment (etiotto) [reply test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir]: done

File: test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir (hunk 6 above)
Comment (etiotto) [reply test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir]: done

File: test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir (hunk 13 above)
Comment (etiotto) [reply test/TritonIntelGPU/accelerate-matmul-edge-cases.mlir:41]: modified

File: test/TritonIntelGPU/accelerate-matmul-int8.mlir (hunk 7 above)
Comment (etiotto) [reply test/TritonIntelGPU/accelerate-matmul-int8.mlir]: done

File: test/TritonIntelGPU/accelerate-matmul-int8.mlir (hunk 8 above)
Comment (etiotto) [reply test/TritonIntelGPU/accelerate-matmul-int8.mlir]: done

File: test/TritonIntelGPU/accelerate-matmul-int8.mlir (hunk 9 above)
Comment (etiotto) [reply test/TritonIntelGPU/accelerate-matmul-int8.mlir]: done

File: test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir (hunk 10 above)
Comment (etiotto) [reply test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir]: RUN lines are collected first and they behave as they were all at the start of the file. I will move the RUN lines at the top to avoid confusion

File: test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir (hunk 11 above)
Comment (etiotto) [reply test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir]: Added `support-2d-block-io=true`. As for the second question, if `threads-per-warp` is omitted the threads-per-warp will default to 32, not 16.

File: test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir (hunk 12 above)
Comment (etiotto) [reply test/TritonIntelGPU/triton-annotate-module-dpas-analysis.mlir]: It tests that in the the min-subgroup-size is annotated correctly
